src/OpenTelemetry.Instrumentation.Process/ProcessMetrics.cs

```diff
@@ -99,5 +142,26 @@ private void Snapshot()
         this.virtualMemoryUsage = this.currentProcess.PrivateMemorySize64;
         this.userProcessorTime = this.currentProcess.UserProcessorTime.TotalSeconds;
         this.privilegedProcessorTime = this.currentProcess.PrivilegedProcessorTime.TotalSeconds;
+        this.cpuUtilization = this.GetCpuUtilization();
+        this.numberOfThreads = this.currentProcess.Threads.Count;
+    }
+
+    private IEnumerable<Measurement<double>> GetCpuUtilization()
+    {
+        var userProcessorUtilization = (this.currentProcess.UserProcessorTime.TotalSeconds - this.collectionUserProcessorTime) /
+                ((DateTime.UtcNow - this.collectionTimeStamp.ToUniversalTime()).TotalSeconds * Environment.ProcessorCount);
+
+        var privilegedProcessorUtilization = (this.currentProcess.PrivilegedProcessorTime.TotalSeconds - this.collectionPrivilegedProcessorTime) /
+                ((DateTime.UtcNow - this.collectionTimeStamp.ToUniversalTime()).TotalSeconds * Environment.ProcessorCount);
+
+        this.collectionTimeStamp = DateTime.UtcNow;
+        this.collectionUserProcessorTime = this.currentProcess.UserProcessorTime.TotalSeconds;
+        this.collectionPrivilegedProcessorTime = this.currentProcess.PrivilegedProcessorTime.TotalSeconds;
+
+        return new[]
+        {
+            new Measurement<double>(userProcessorUtilization, new KeyValuePair<string, object?>("state", "user")),
+            new Measurement<double>(privilegedProcessorUtilization, new KeyValuePair<string, object?>("state", "system")),
```

Comment:
Maybe we should use Math.Min here to make sure it never reports above 1 (100%) utilization?

```csharp
            new Measurement<double>(Math.Min(userProcessorUtilization, 1d), new KeyValuePair<string, object?>("state", "user")),
            new Measurement<double>(Math.Min(privilegedProcessorUtilization, 1d), new KeyValuePair<string, object?>("state", "system")),
```

Reply:
updated